### PR TITLE
Support relative DID URLs for verification method IDs

### PR DESCRIPTION
**Pull request by reinkrul**
Implements https://www.w3.org/TR/did-core/#relative-did-urls

Verification method IDs can be relative to their document DID. The same applies to relationships.

It also applies to service IDs, but those are URIs (rather than DID URLs), so those already parsed correctly (and we don't do any resolving on the IDs in this library).

Required for supporting SIOPv2/OpenID4VP flow with Microsoft Authenticator

### CODE REVIEW

**Comment by reinkrul** [comment]
See https://github.com/nuts-foundation/go-did/issues/88

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
When you do ParseDID, there must be a `did` prefix. But when parsing a key ID you create a DID without prefix and/or method. I think a DIDUrl type needs to be introduced that is used in most of the methods (and refer to the DID) and that will keep the DID "pure"

**Comment by reinkrul** [comment]
Introduced `DIDURL` type, which makes things more pure. Only used for `VerificationMethod.id`, but will probably be need to changed here and there in nuts-node.

File: did/didurl.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
+package did
```
Comment (woutslakhorst) [inline did/didurl.go:1]: copyright

**Review by woutslakhorst — APPROVED** [review]

File: did/didurl.go (hunk 1 above)
Comment (reinkrul) [reply did/didurl.go:1]: None of the files in this lib have a copyright notice. We should add it in a separate PR.

**Pull request merged**